### PR TITLE
Filebrowser icons

### PR DESCRIPTION
**Pull request by tomrudnick**
Second Attempt for FileBrowser Icons
I added the Filebrowser Icons from Devicon to our project.
For some reason, the Devicon submodule was only working for me
I included them now in the Assets.

Nearly all icons are available in two styles: .plain and .original
if an icon is not available in one style it will pick the other one.

IMG_0DCC4D7EBA82-1

### CODE REVIEW

File: Themis/Models/Repository.swift [hunk 1]
```diff
@@ -80,7 +101,26 @@ class Node: Hashable, ObservableObject {
     var path: String {
         calculatePath().joined(separator: "/")
     }
-
+    
+    private let languageDict: [String: Language] = [
+        "c" : .c,
+        "py" : .python,
```
Comment (pal03377) [inline Themis/Models/Repository.swift]: Fun fact: There's also the `.pyw` file extension, which is also associated with Python (for running without a Terminal). We might want to include that as well.
```suggestion
        "py" : .python,
        "pyw": .python,
```

File: Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift [hunk 2]
```diff
@@ -46,4 +50,39 @@ struct FiletreeSidebarView: View {
...
+            return Image(uiImage: image)
+        }
+        let alternativeStyle = style == .plain ? Style.original : .plain
+        let alternativeImageName = "\(language.rawValue)-\(alternativeStyle.rawValue)"
+        if let image = UIImage(named: alternativeImageName) {
+            return Image(uiImage: image)
+        }
+        return nil
+    }
+
```
Comment (pal03377) [inline Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift]: I feel like there's enough logic in here that would justify moving this into its own component => own file.

**Review by pal03377 — CHANGES_REQUESTED** [review]
Considering we only support icons on a few file types anyways, I think we should drop all of the unnecessary icons from our assets folder and only add documentation on how to add new languages somewhere. Including all of those icons increases our compressed app size from 1.5MB to 6.6MB, which is a bit much.

[App Thinning Size Report old.txt](https://github.com/ls1intum/Themis/files/10479815/App.Thinning.Size.Report.old.txt)
[App Thinning Size Report new.txt](https://github.com/ls1intum/Themis/files/10479814/App.Thinning.Size.Report.new.txt)

**Comment by tomrudnick** [comment]
> Considering we only support icons on a few file types anyways, I think we should drop all of the unnecessary icons from our assets folder and only add documentation on how to add new languages somewhere. Including all of those icons increases our compressed app size from 1.5MB to 6.6MB, which is a bit much.
> 
> [App Thinning Size Report old.txt](https://github.com/ls1intum/Themis/files/10479815/App.Thinning.Size.Report.old.txt) [App Thinning Size Report new.txt](https://github.com/ls1intum/Themis/files/10479814/App.Thinning.Size.Report.new.txt)

Ok, so everyone should provide me with a list of unnecessary programming languages/file types, and I will take the intersection of this list and remove those icons. Or, what do you propose to remove "unnecessary" file types? 😂. I think it would be much more appropriate to add more file types from time to time and add them to the dictionary, and to be happy that the icon is probably already there. I don't think someone will decide not to use this app because the file size is now 5 MB larger. But of course, I can remove icons if you guys think they are unnecessary, just provide a list.

**Comment by pal03377** [comment]
> I don't think someone will decide not to use this app because the file size is now 5 MB larger

Of course, that's true. However, I think that just increasing the app size threefold just for the potential of a bit of future development convenience is overkill as well.

Here's a list of languages that I either never heard of or that I cannot imagine being relevant in a programming exercise:
```
aarch64
adonisjs
apachekafka
appcelerator
appwrite
argocd
atom
azure
backbonejs
bamboo
bitbucket
bootstrap
bower
bulma
cakephp
centos
ceylon
circleci
clojure
clojurescript
codecov
codeigniter
composer
couchdb
cplusplus
crystal
csharp
cucumber
d3js
debian
denojs
devicon
digitalocean
discordjs
doctrine
dot-net
dotnetcore
dropwizard
drupal
eleventy
elixir
elm
embeddedc
ember
erlang
feathersjs
fedora
filezilla
firebase
foundation
fsharp
gatling
gatsby
gentoo
gitter
google
googlecloud
grafana
grails
handlebars
haxe
heroku
hugo
ie10
ionic
jamstack
jeet
jenkins
k3s
kaggle
karma
knockout
krakenjs
kubernetes
labview
laravel
less
linux
magento
materialui
meteor
mocha
modx
msdos
neo4j
nestjs
networkx
nextjs
nodewebkit
nuget
nuxtjs
openal
opensuse
opera
packer
pandas
phalcon
phoenix
phpstorm
podman
polygon
prometheus
protractor
pycharm
qt
rect
redhat
rocksdb
sdl
selenium
sequelize
shopware
shotgrid
socketio
solidity
spss
storybook
stylus
subversion
symfony
thealgorithms
tomcat
tortoisegit
towergit
ubuntu
unix
vagrant
vim
vuestorefront
vuetify
webflow
weblate
webstorm
woocommerce
xamarin
xd
yii
yunohost
zend
zig
```

File: Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift (hunk 2 above)
Comment (tomrudnick) [reply Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift]: Ok I extracted it

**Comment by tomrudnick** [comment]
> > I don't think someone will decide not to use this app because the file size is now 5 MB larger
> 
> Of course, that's true. However, I think that just increasing the app size threefold just for the potential of a bit of future development convenience is overkill as well.
> 
> Here's a list of languages that I either never heard of or that I cannot imagine being relevant in a programming exercise:
> 
Ok Yeah I removed those files you are right

**Review by pal03377 — APPROVED** [review]
Well done!

**Review by katjanakosic — APPROVED** [review]
Looks wayyy nicer now that the icons are added 😁

**Review by huber-florian — APPROVED** [review]
Thanks for all your effort. What is the current plan? Removing or keeping "unnecessary" icons?

**Comment by tomrudnick** [comment]
I probably going to remove some more icons that are clearly not going to be in a Filetype in an exercise and from time to time add more file types, furthermore merge those dicts with the code editor language Themes, probably discuss that on Wednesday

**Pull request merged**